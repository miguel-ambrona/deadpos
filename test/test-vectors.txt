// Josef JÃ¡na, 1928 (Neues Grazer Tagblatt), 1 solution.
r3k3/3N4/2N5/8/1K6/8/7p/8 b q - >>= h#3 >>= legal

// Chris Feather, 1975 (Schach), 2 solutions.
1RrB2b1/8/4n3/2n3p1/2K2b2/1p1rk3/6BR/8 b - - >>= h#2 >>= legal

// Milan Vukcevich, 1996 (CHM avec 6 pieces Bad Pyrmont), 1 solution.
4K2n/3P1P2/4k3/3p4/8/8/8/8 b - - >>= h#2 >>= legal

// Roddy McKay, 2021 (Matplus.net), 1 solution.
8/1B6/3p1P1q/4ppB1/1p3b2/1Pk3p1/2P4p/1K6 b - - >>= h#2 >>= legal

// Neal Turner, 1995 (Problemist Supplement), 1 solution.
q1b1n1K1/2P1P3/8/3k4/8/8/8/8 b - - >>= h#3 >>= legal

// Miguel Ambrona, 2021 (Matplus.net), 1 solution.
8/4K2k/4P2p/8/5q2/2b5/8/8 b - - >>= h#3 >>= legal
8/8/2B5/5Q2/8/4p2P/4k2K/8 w - - >>= h#3 half-duplex >>= legal

// Test en-passant
4k2r/8/2P1P1P1/2PpKpP1/8/8/8/8 w k ? >>= legal >>= ep

// Test castling
4k2r/8/8/8/8/8/8/4K3 w >>= legal >>= castling
