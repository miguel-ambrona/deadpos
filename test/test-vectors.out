Deadpos Analyzer version 2.3
// Josef JÃ¡na, 1928 (Neues Grazer Tagblatt), 1 solution.
>>> r3k3/3N4/2N5/8/1K6/8/7p/8 b q - >>= h#3 >>= legal
h2h1b c6b8 h1b7 b8a6 e8c8 d7b6# 2kr4/1b1N4/N7/8/1K6/8/8/8 w - - 4 4             
nsols 1                                                                        

// Chris Feather, 1975 (Schach), 2 solutions.
>>> 1RrB2b1/8/4n3/2n3p1/2K2b2/1p1rk3/6BR/8 b - - >>= h#2 >>= legal
f4b8 g2d5 e6c7 d8g5# 1brB2b1/2n5/8/2nB2p1/2K5/1p1rk3/7R/8 w - - 2 3             
d3d8 g2c6 c5d7 b8b3# 1Rrr2b1/3n4/2B1n3/6p1/2K2b2/1p2k3/7R/8 w - - 2 3           
nsols 2                                                                        

// Milan Vukcevich, 1996 (CHM avec 6 pieces Bad Pyrmont), 1 solution.
>>> 4K2n/3P1P2/4k3/3p4/8/8/8/8 b - - >>= h#2 >>= legal
h8g6 f7f8q g6e5 d7d8n# 4KQ2/3P4/4k3/3pn3/8/8/8/8 w - - 1 3                      
nsols 1                                                                        

// Roddy McKay, 2021 (Matplus.net), 1 solution.
>>> 8/1B6/3p1P1q/4ppB1/1p3b2/1Pk3p1/2P4p/1K6 b - - >>= h#2 >>= legal
f4c1 g5c1 h6d2 c1b2# 8/1B6/3p1P2/4pp2/1p6/1Pk3p1/2Pq3p/1KB5 w - - 1 3           
nsols 1                                                                        

// Neal Turner, 1995 (Problemist Supplement), 1 solution.
>>> q1b1n1K1/2P1P3/8/3k4/8/8/8/8 b - - >>= h#3 >>= legal
c8f5 c7c8r d5e6 c8e8 a8d8 e7d8q# 3qR1K1/4P3/4k3/5b2/8/8/8/8 w - - 1 4           
nsols 1                                                                        

// Miguel Ambrona, 2021 (Matplus.net), 1 solution.
>>> 8/4K2k/4P2p/8/5q2/2b5/8/8 b - - >>= h#3 >>= legal
f4b8 e7f7 c3h8 e6e7 b8f8 e7f8n# 5q1b/4PK1k/7p/8/8/8/8/8 w - - 1 4               
nsols 1                                                                        

>>> 8/8/2B5/5Q2/8/4p2P/4k2K/8 w - - >>= h#3 half-duplex >>= legal
f5b1 e2f2 c6h1 e3e2 b1f1 e2f1n# 8/8/8/8/8/7P/4pk1K/5Q1B b - - 1 3               
nsols 1                                                                        

// Test en-passant
>>> 4k2r/8/2P1P1P1/2PpKpP1/8/8/8/8 w k ? >>= legal >>= ep
d6
f6
nsols 2                                                                        

// Test castling
>>> 4k2r/8/8/8/8/8/8/4K3 w >>= legal >>= castling
-
nsols 1                                                                        

